backend/src/common/plugins/socket-injector.plugin.ts

```diff
@@ -0,0 +1,22 @@
+import fastifyPlugin from 'fastify-plugin';
+import { type Server as SocketServer } from 'socket.io';
+
+import { PluginName } from '~/common/enums/enums.js';
+
+type Options = {
+    io: SocketServer;
+};
+
+const socketInjectorPlugin = fastifyPlugin<Options>(
+    (fastify, { io }, done) => {
+        fastify.decorateRequest('io', null);
+        // eslint-disable-next-line @typescript-eslint/require-await
+        fastify.addHook('preHandler', async (request) => {
```

Comment:
Should this fn should be async?

Reply:
I double checked, yeah, we can use done() to avoid async. I fixed it